### PR TITLE
Implement Backend Error Redesign

### PR DESCRIPTION
**Pull request by alcalawil**
[Docs Link](https://www.notion.so/stringxyz/API-Errors-2ac0063307994d369e10a423efc839ab?pvs=4)

### CODE REVIEW

File: httperror/http_error.go [hunk 1]
```diff
@@ -33,53 +31,51 @@ func BadRequestError(c echo.Context, message ...string) error {
...
-	return c.JSON(http.StatusUnauthorized, JSONError{Message: "Unauthorized", Code: "UNAUTHORIZED"})
+	return c.JSON(http.StatusUnauthorized, JSONError{Message: "This action requires authentication", Code: "UNAUTHORIZED"})
 }
 
-func TokenExpired(c echo.Context, message ...string) error {
-	msg := "Token Expired"
+func InternalError(c echo.Context, message ...string) error {
 	if len(message) > 0 {
-		msg = strings.Join(message, " ")
+		return c.JSON(http.StatusInternalServerError, JSONError{Message: strings.Join(message, " "), Code: "INTERNAL_SERVER"})
```
Comment (saito-sv) [inline httperror/http_error.go:64]: Nit: Optional. 
```suggestion
		return c.JSON(http.StatusInternalServerError, JSONError{Message: strings.Join(message, " "), Code: "INTERNAL})
```

**Review by saito-sv — APPROVED** [review]
LGTM

File: database/redis.go [hunk 2]
```diff
@@ -7,6 +7,8 @@ import (
 	"time"
 
 	"github.com/String-xyz/go-lib/common"
+	serrors "github.com/String-xyz/go-lib/stringerror"
```
Comment (ocasta181) [inline database/redis.go:10]:
```suggestion
	serror "github.com/String-xyz/go-lib/stringerror"
```

File: database/redis.go [hunk 3]
```diff
@@ -118,6 +120,11 @@ func (r redisStore) Get(id string) ([]byte, error) {
 	ctx := context.Background()
 	bytes, err := r.client.Get(ctx, id).Bytes()
 	if err != nil {
+
+		if err.Error() == REDIS_NOT_FOUND_ERROR {
+			return nil, common.StringError(serrors.NOT_FOUND)
```
Comment (ocasta181) [inline database/redis.go:125]:
```suggestion
			return nil, common.StringError(serror.NOT_FOUND)
```

File: database/redis.go [hunk 4]
```diff
@@ -144,6 +151,11 @@ func (r redisStore) HGetAll(key string) (map[string]string, error) {
 	ctx := context.Background()
 	data, err := r.client.HGetAll(ctx, key).Result()
 	if err != nil {
+
+		if err.Error() == REDIS_NOT_FOUND_ERROR {
+			return nil, common.StringError(serrors.NOT_FOUND)
```
Comment (ocasta181) [inline database/redis.go:156]:
```suggestion
			return nil, common.StringError(serror.NOT_FOUND)
```

File: repository/repository.go [hunk 5]
```diff
@@ -8,6 +8,8 @@ import (
 	"strings"
 	"time"
 
+	serrors "github.com/String-xyz/go-lib/stringerror"
```
Comment (ocasta181) [inline repository/repository.go:11]:
```suggestion
	serror "github.com/String-xyz/go-lib/stringerror"
```

File: repository/repository.go [hunk 6]
```diff
@@ -61,15 +63,15 @@ func (b Base[T]) List(ctx context.Context, limit int, offset int) (list []T, err
...
+		return list, nil
 	}
 	return list, err
 }
 
 func (b Base[T]) GetById(ctx context.Context, ID string) (m T, err error) {
 	err = b.Store.GetContext(ctx, &m, fmt.Sprintf("SELECT * FROM %s WHERE id = $1 AND deactivated_at IS NULL", b.Table), ID)
 	if err != nil && err == sql.ErrNoRows {
-		return m, err
+		return m, serrors.NOT_FOUND
```
Comment (ocasta181) [inline repository/repository.go:74]:
```suggestion
		return m, serror.NOT_FOUND
```

File: repository/repository.go [hunk 7]
```diff
@@ -78,7 +80,7 @@ func (b Base[T]) GetById(ctx context.Context, ID string) (m T, err error) {
 func (b Base[T]) GetByUserId(ctx context.Context, userID string) (m T, err error) {
 	err = b.Store.GetContext(ctx, &m, fmt.Sprintf("SELECT * FROM %s WHERE user_id = $1 AND deactivated_at IS NULL LIMIT 1", b.Table), userID)
 	if err != nil && err == sql.ErrNoRows {
-		return m, err
+		return m, serrors.NOT_FOUND
```
Comment (ocasta181) [inline repository/repository.go:83]:
```suggestion
		return m, serror.NOT_FOUND
```

**Review by ocasta181 — CHANGES_REQUESTED** [review]

File: httperror/http_error.go (hunk 1 above)
Comment (alcalawil) [reply httperror/http_error.go:64]: later reftactor

**Review by ocasta181 — APPROVED** [review]

**Pull request merged**